src/solver/conv_asm_1x1u.cpp

```diff
@@ -243,75 +244,182 @@ bool PerformanceConfigConvAsm1x1U::operator==(const PerformanceConfigConvAsm1x1U
 
 bool PerformanceConfigConvAsm1x1U::IsValidValue() const
 {
-    // clang-format off
-    return IsLinear<1,4>(read_size)
-        && Is_1_4_8_12_to_32(k_mult)
-        && IsLinear<1,16>(chunks_per_wave)
-        && IsTwoPower<1,64>(chunk_size)
-        && IsLinear<1,8>(n_mult)
-        && IsTwoPower<1,32>(c_mult)
-        && IsLinear<1,8>(waves_c_in_group)
-        && IsTwoPower<1,8>(waves_k_in_group); // clang-format on
+    bool is_valid_value = true;
+    if(read_size != -1)
```

Comment:
BTW why do we need to assume that `-1` is valid value here?

Reply:
I've made both IsValid() and IsValidValue() functions "dynamic", meaning that they can take partially initialized performance configs and check for validity without outright failing if one of the tuning parameters is the default value. This is needed because the model produces the tuning parameter sequence one "token" or value at a time and we need to verify if the value is valid for every "token" produced. The value also needs to be valid with the sequence produced so far. If it is not then we take the next most likely "token" from the vector of probabilities the model produces. (The "token" being the index of the highest probability in the output vector if not valid then the next index with highest probability, etc.). The value is gotten from using the "token" as the key in the decodings map.

Reply:
I have added the "dynamic" functions in addiction to default functions.